### PR TITLE
PDI-13367 - ETL Metada Injection does not actualize the Source Step in the Inject Metadata tab when renaming the cited step

### PR DESCRIPTION
**Pull request by yuryilyukevich**

### CODE REVIEW

**Comment by yuryilyukevich** [comment]
@mattyb149 Please, look it.

File: engine/src/org/pentaho/di/trans/step/StepMetaChangeListenerInterface.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package org.pentaho.di.trans.step;
```
Comment (mattyb149) [inline engine/src/org/pentaho/di/trans/step/StepMetaChangeListenerInterface.java]: Needs a copyright header

File: engine/src/org/pentaho/di/trans/step/StepMetaChangeListenerInterface.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import org.pentaho.di.trans.TransMeta;
+
+public interface StepMetaChangeListenerInterface {
+  /**
+   * 
+   *
```
Comment (mattyb149) [inline engine/src/org/pentaho/di/trans/step/StepMetaChangeListenerInterface.java]: Please add javadoc here

**Comment by yuryilyukevich** [comment]
@mattyb149 The fix was updated.

**Comment by kurtwalker** [comment]
@mattyb149 @yuryilyukevich Looks like this change has caused an error in all spoon dialogs.  Open a step dialog in any transformation, then click ok.  You'll get an exception complaining about notify listeners.

**Pull request merged**